### PR TITLE
gh-101747: Fix refleak in new `OrderedDict` repr

### PR DESCRIPTION
**Pull request by sobolevn**
After: `./python.exe -m test -v test_ordered_dict -R 3:3`

```
Ran 279 tests in 1.460s

OK
.

== Tests result: SUCCESS ==

1 test OK.

Total duration: 10.2 sec
Tests result: SUCCESS
```

* Issue: gh-101747

### CODE REVIEW

**Review by kumaraditya303 — APPROVED** [review]

**Review by erlend-aasland — APPROVED** [review]

**Comment by erlend-aasland** [comment]
The `aarch64 RHEL8` refleak bot still reports a leak. Could you please examine, @sobolevn?

File: Objects/odictobject.c
Comment (serhiy-storchaka) [inline Objects/odictobject.c]: You can move it above the `Done` label and change `Py_XDECREF` to `Py_DECREF`.

**Comment by sobolevn** [comment]
@erlend-aasland it reports:

```
test_importlib leaked [6, 4, 2] references, sum=12
test_importlib leaked [4, 4, 2] memory blocks, sum=10
```

I think these leaks are not related 🤔 

Should I fix it here?

File: Objects/odictobject.c
Comment (sobolevn) [reply Objects/odictobject.c]: Thanks, it might be a slightly more readable! I will wait for all tests to finish and apply it :)

**Comment by erlend-aasland** [comment]
> Should I fix it here?

I prefer one fix per PR; that makes it easier to bisect in case something goes wrong, and it also makes for a cleaner git history.

**Comment by sobolevn** [comment]
@erlend-aasland then I consider this done:
1. Review suggestion is applied
2. New issue is created: https://github.com/python/cpython/issues/101766

**Review by corona10 — APPROVED** [review]
lgtm

**Comment by corona10** [comment]
I will merge this PR after waiting for @serhiy-storchaka approval

> test_importlib leaked [6, 4, 2] references, sum=12

I can not reproduce the issue on my macOS. It sounds like we have to debug on the RHEL machine.

**Review by rhettinger — APPROVED** [review]

**Pull request merged**
